include/ScratchViews.h

```diff
@@ -484,16 +503,27 @@ ScratchViews<T>::ScratchViews(const TeamHandleType& team,
   int numScvIp = meSCV != nullptr ? meSCV->numIntPoints_ : 0;
   int numFemIp = meFEM != nullptr ? meFEM->numIntPoints_ : 0;
 
-  create_needed_field_views(team, dataNeeded, bulkData, nodesPerEntity);
+  create_needed_field_views(team, dataNeeded, bulkData, nodalGatherSize);
 
   create_needed_master_element_views(team, dataNeeded, nDim, nodesPerFace, nodesPerElem, numFaceIp, numScsIp, numScvIp, numFemIp);
 }
 
+template<typename T>
+ScratchViews<T>::ScratchViews(const TeamHandleType& team,
```

Comment:
What is the motivation for a second constructor here? It is only used for elemViews?

Reply:
The complexity here is that for a face or interior algorithm, we have all we need, e.g., meFC or meSCS. Moreover, the nodesPerEntity is uniquely defined by either nodesPerFace or nodesPerElement. For the face/element combination, the developer needs are more complex. For example, we might need both meFC and meSCS. Worse yet, an exposed face grad op requires meSCS->face_grad_op(). However, this is sized by numFacIp*nodesPerElem*nDim. However, meSCS has not interface to providing numFaceIp for each ordinal. The best path forward, at least for now, seemed to be to create a new ScratchViews constructor with all of the information packed up in a struct.

Reply:
@spdomin Thanks for the explanation.

Reply:
There is a lot of nitty gritty here. Let's meet as a team next week and talk through some of the current complexities and how to move forward. At present, we are managing face/elem, face, and elem data through one ElemDataRequest type. We seem to go back and forth with considering breaking this into two data requests types. Also, nodal gathering is somewhat independent of master element calls with exception of shared nodal fields - solely "coordinates".

The plan is to start iterating on how to make things more clean. In fact, it may be fine now, we just need more experience with writing face and face/elem kernels along with the accompanied FaceElemSolverAlg (@alanw0, you can start on that this morning and also process this pull request if it looks consistent with what we were doing the other day).

At the very least, this allows us to move forward with all of the other required pieces to start a full face/elem kernel approach. My plan is to deprecate the symmetry bc ASAP.

Reply:
@sayerhs Regarding the second constructor for ScratchViews: we also discussed the possibility of creating a DataRequests base-class, with ElemDataRequests and FaceDataRequests derived classes. That would still require an overloaded constructor etc for ScratchViews. We can keep iterating on the best design as we gain more experience with face-element algorithms.

@spdomin I'll work on elevating the faceElemAlg class from the unit-test to a "real" AssembleFaceElemSolverAlgorithm class.